tests/integration.rs

```diff
@@ -0,0 +1,433 @@
+use std::collections::HashMap;
+use std::error::Error as StdError;
+use std::io::Read;
+use std::net::TcpStream;
+use std::process::{Child, Command};
+use std::sync::LazyLock;
+use std::time::Duration;
+use std::{env, fs, io, thread};
+
+use bytes::{Buf, Bytes};
+use http::header::CONTENT_TYPE;
+use http::{Method, Request};
+use http_body_util::{BodyExt, Full};
+use hyper_util::client::legacy::connect::HttpConnector;
+use hyper_util::client::legacy::Client as HyperClient;
+use hyper_util::rt::TokioExecutor;
+use instant_acme::{
+    Account, AuthorizationStatus, ChallengeType, Identifier, NewAccount, NewOrder, Order,
+    OrderStatus,
+};
+use rcgen::{CertificateParams, DistinguishedName, KeyPair};
+use rustls::client::{verify_server_cert_signed_by_trust_anchor, verify_server_name};
+use rustls::crypto::CryptoProvider;
+use rustls::pki_types::pem::PemObject;
+use rustls::pki_types::{CertificateDer, ServerName};
+use rustls::server::ParsedCertificate;
+use rustls::RootCertStore;
+use rustls_pki_types::UnixTime;
+use serde::Serialize;
+use serde_json::json;
+use tempfile::NamedTempFile;
+use tokio::time::sleep;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn StdError>> {
+    // Spawn a Pebble CA and a challenge response server.
+    let pebble = PebbleEnvironment::new(DEFAULT_CONFIG.clone())?;
+
+    // Create a test account with the Pebble CA.
+    let client = Box::new(pebble.http_client());
+    let (account, _) = Account::create_with_http(
+        &NewAccount {
+            contact: &[],
+            terms_of_service_agreed: true,
+            only_return_existing: false,
+        },
+        &pebble.directory_url(),
+        None,
+        client,
+    )
+    .await?;
+
+    // Create an order.
+    let identifier = Identifier::Dns("example.com".to_string());
+    let order = account
+        .new_order(&NewOrder {
+            identifiers: &[identifier],
+        })
+        .await?;
+
+    // Issue a certificate w/ HTTP-01 challenge and verify it.
+    test_http1(&pebble, order).await?;
+
+    Ok(())
+}
+
+async fn test_http1(env: &PebbleEnvironment, mut order: Order) -> Result<(), Box<dyn StdError>> {
```

Comment:
Maybe this should be a `PebbleEnvironment` method?

Reply:
I moved this and a handful of other free-standing functions to be methods of the environment.